wpsc-components/merchant-core-v3/libraries/amazon-payments/assets/js/amazon-checkout.js

```diff
@@ -0,0 +1,39 @@
+jQuery(function() {
+
+	// Login Widget
+	new OffAmazonPayments.Widgets.Button ({
```

Comment:
I think i found the issue. You are using the OffAmazonPayments.Widgets.Button constructor.
Newest interface is just OffAmazonPayments.Button. 
You will need to set the client id in the head as well.
<head>
  <script type='text/javascript'>
    window.onAmazonLoginReady = function() {
      amazon.Login.setClientId('YOUR_CLIENT_ID_HERE');
    };
  </script>
  <script type='text/javascript'
src='https://static-na.payments-amazon.com/OffAmazonPayments/us/js/Widgets.js?sellerId=YOUR_SELLER_ID_HERE'>
  </script>
</head>

<script type="text/javascript">
  var authRequest;
  OffAmazonPayments.Button("AmazonPayButton", "YOUR_SELLER_ID_HERE", {
    type:  "PwA",
    color: "Gold",
    size:  "medium",
    useAmazonAddressBook: true,
    authorization: function() {
      var loginOptions = {scope: 'profile payments:widget'};
      authRequest = amazon.Login.authorize(loginOptions, "YOUR_REDIRECT_URL_HERE");
    },
    onError: function(error) {
      // Write your custom error handling
    }
  });
</script>